app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java

```diff
@@ -79,154 +102,169 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
-//        //Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
-//        displayLocation();
-//    }
-//
-//    public void onSaveInstanceState(Bundle savedInstanceState) {
-//        savedInstanceState.putParcelable(KEY_LOCATION, mCurrentLocation);
-//        super.onSaveInstanceState(savedInstanceState);
-//    }
-//
-//    @Override
-//    public void onMapLongClick(@NonNull LatLng latLng) {
-//
-//    }
+        map = googleMap;
+        getLocation();
+//        googleMap.addMarker(new MarkerOptions()
+//                .position(new LatLng(51.5072, 0.1276))
+//                .title("London"));
+//        LatLng latLng = new LatLng(51.5072, 0.1276);
+//        map.moveCamera(CameraUpdateFactory.newLatLng(latLng));
+//        map.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng,15f));
+    }
+
+    private void getLocation() {
+        progressDialog.show();
+        if(Build.VERSION.SDK_INT>=23){
+            if(checkPermission()){
+                getDeviceLocation();
+            }
+            else{
+                requestPermission();
+            }
+        }
+        else{
+            getDeviceLocation();
+        }
+    }
+
+    private void requestPermission() {
+        ActivityCompat.requestPermissions(getActivity(),permissions_all,PERMISSION_CODE);
+    }
+
+    private boolean checkPermission() {
+        for(int i=0;i<permissions_all.length;i++){
+            int result= ContextCompat.checkSelfPermission(getContext(),permissions_all[i]);
+            if(result== PackageManager.PERMISSION_GRANTED){
+                continue;
+            }
+            else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private void getDeviceLocation() {
+        locationManager=(LocationManager) getActivity().getSystemService(Service.LOCATION_SERVICE);
+        isGpsLocation=locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+        isNetworklocation=locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+        if(!isGpsLocation && !isNetworklocation){
+            showSettingForLocation();
+            getLastlocation();
+        }
+        else{
+            getFinalLocation();
+        }
+    }
+
+    private void getLastlocation() {
+        if(locationManager!=null) {
+            try {
+                Criteria criteria = new Criteria();
+                String provider = locationManager.getBestProvider(criteria,false);
+                Location location=locationManager.getLastKnownLocation(provider);
+            } catch (SecurityException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        switch (requestCode){
+            case PERMISSION_CODE:
```

Comment:
This is what we would call a "code smell"...
https://en.wikipedia.org/wiki/Code_smell

While it may be functionally correct code, a single case in a switch implies you meant to have more of them which are missing, which would suggest a potential issue. This code would be more succinct if you just made this a if statement